presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -328,6 +330,26 @@ else if (temporalType.equals(TIMESTAMP)) {
         return builder.build();
     }
 
+    private static void buildPredicatesAndValues(Domain nameDomain, List<String> values, List<String> predicates, String predicatePrefix)
+    {
+        if (nameDomain.isSingleValue()) {
+            predicates.add(predicatePrefix + " = ?");
+            values.add(getStringValue(nameDomain.getSingleValue()));
+        }
+        else {
+            List<String> names = nameDomain.getValues().getRanges().getOrderedRanges().stream()
+                    .filter(Range::isSingleValue)
+                    .map(range-> getStringValue(range.getSingleValue()))
+                    .collect(toList());
+            // Add a predicate only when nameDomain has actual content. Cases like "select * from system.shards where table_name <'ab';" will be ingnored.
```

Comment:
Typo "ingnored"

The comment is a bit confusing. I'm not sure what "actual content" means. I think we can remove it entirely, as it seems fairly obvious what is happening (based on the generated SQL)